### DISCUSSION TITLE
Have LuckPerms be automatically downloaded to supersed permissions.yml file.

### DISCUSSION THREAD

**Discussion by Kichura** [PurpurMC/Purpur#1276] [__Dismissed__] [answered]
Currently by default without any plugins present; 

Purpur re-uses the same permissions.yml system that users could use themselves to manage permissions which can end up redundant to some extent, Instead what purpur could provide is automatically download LuckPerms as a way to replace the old bukkit permissions system similar to what Spark did when the timings system was superseded in it's patches list,

for addition, this system can also be unsubscribed from using a JVM argument statement in case LuckPerms does not seem to fitting for the server owner itself for any given reasoning.

**Comment by RealRONiN** [comment]
A good amount of people prefer other permission plugins over Luckperms, and most people who use shared hostings don't get to change JVM args.

**Reply by Kichura** [reply]
For the latter; This isn't always the case such as bisecthosting and other services i have seen so far but can see how that would be bothering to resolve otherwise if there is basically no way to cancel luckperms from being automatically present along with spark in terms of management,

As for the former; that one doesn't really seem to add up since you could always migrate them to the new plugin once PEX *(for example)* reaches end-of-life at some point in future, although i would like to hear about the reasoning why luckperms isn't adapted as much by those people to begin with.

**Reply by RealRONiN** [reply]
> For the latter; This isn't always the case such as bisecthosting and other services i have seen so far but can see how that would be bothering to resolve otherwise if there is basically no way to cancel luckperms from being automatically present along with spark in terms of management,

Exactly my point, it isn't always the case with 'most' hosting sites, not everyone provide access to JVM flags, some just provide a simple toggle button to add Aikar's flags even.

> As for the former; that one doesn't really seem to add up since you could always migrate them to the new plugin once PEX (for example) reaches end-of-life at some point in future, although i would like to hear about the reasoning why luckperms isn't adapted as much by those people to begin with.

If the server detects that there's another plugin installed for permission management or somehow can coexist with one another, then that's acceptable, but still isn't worth it imho.

**Comment by katekerllenevich** [comment, marked as answer]
For reasons mentioned on other comments, we aren't going to do this. The only reason we bundle spark is because we needed a replacement to timings and people who are desperate for support are going to have a hard time installing a plugin (sometimes). We don't like to bundle in plugins if we don't have to, and especially not permissions as some servers don't even use it, or use a different plugin. Also we couldn't make it opt-out with a JVM argument as many people don't have access to their JVM arguments.
